packages/blocks/src/lib/navigation-map/stores.ts

```diff
@@ -0,0 +1,27 @@
+import * as THREE from 'three';
+import { writable } from 'svelte/store';
+import { currentWritable } from '@threlte/core';
+import type { Waypoint, Obstacle } from '$lib';
+
+export const hovered = currentWritable<string | null>(null);
+
+export const waypoints = currentWritable<Waypoint[]>([]);
+export const obstacles = currentWritable<Obstacle[]>([]);
+
+/** The currently selected tab. */
+export const tab = writable<'Obstacles' | 'Waypoints'>('Waypoints');
```

Comment:
I reeaaallly dislike enums in TS 😭 and very much prefer string unions. I don't see an advantage to enums, and do see a lot of unnecessary added verbosity and noise. With string unions type checking is just as strict, and you have an equal amount of issues with having to find/replace if you decide your naming was poor. But I might have not yet heard the right argument for enums?

Reply:
That's fair, in that case, does `Record<string, string> as const` fit needs? I find that they're more flexible than enums:

```ts
export const TabType = {
  Obstacles: 'obstacles',
  Waypoints: 'waypoints',
} as const

// This way we get a string union automatically from the object 
export type TabTypes = keyof typeof TabType
```

Reply:
> ```ts
> export const TabType = {
>   Obstacles: 'Obstacles',
>   Waypoints: 'Waypoints',
> } as const
> ```

I'm down to use const objects instead of enums, I've just been finding enums a little more convenient, since we don't need to both defined the object and and the type. The `TabType` (object) vs `TabTypes` (union type) feels a little duplicative and confusing

> I find that they're more flexible than enums:

Since I've only just started using enums, I'd be curious to hear about what sort of inflexibility you've encountered

> ```ts
> // This way we get a string union automatically from the object 
> export type TabTypes = keyof typeof TabType
> ```

This forces us to write an object that has matching keys and values, which could lead to annoyances. I think this would need to be a union on the values, instead:

```ts
export type TabTypes = typeof TabType[keyof typeof TabType]
```

Reply:
Question: how would you solve this problem using enums:

```
let store = writable<'Waypoints' | 'Obstacles'>('Waypoints')
```

Would it just be?

```
const enum Tabs {
  Waypoints = 'Waypoints',
  Obstacles = 'Obstacles',
}

export const tab = writable<Tabs.Waypoints | Tabs.Obstacles>(Tabs.Waypoints);
```

Or is there a better way? Don't love the above but happy to update this to be consistent if the rest of our code is moving in this direction

Reply:
> Question: how would you write this block using enums:
> 
> ```ts
> let store = writable<'Waypoints' | 'Obstacles'>('Waypoints')
> ```

You can use the enum name itself the same way you would use a union type:

```ts
const enum Tab {
  Waypoints = 'Waypoints',
  Obstacles = 'Obstacles',
}

export const tab = writable<Tab>(Tab.Waypoints);
```

> One last thing: if we go with enums I'd prefer it to remain internal, this to me looks like an overcomplicated public api:
> 
> ```ts
> import { NavigationMap, Tabs } from '@viamrobotics/prime-blocks'
> 
> ...
> <NavigationMap tab={Tabs.Obstacles} />
> ```
> 
> I'd rather
> 
> ```ts
> import { NavigationMap, Tabs } from '@viamrobotics/prime-blocks'
> 
> ...
> <NavigationMap tab='Obstacles' />
> ```

Matter of personal preference; I prefer the first one 😝. In order to accept a string-based public API  - which has its benefits! - then we should go with the object-based approach above. That would give us the best of both worlds, though naming things becomes ever-so-slightly annoying

```ts
export const NavigationTab = {
  Waypoints: 'Waypoints',
  Obstacles: 'Obstacles',
} as const

export type NavigationTabType = typeof NavigationTab[keyof typeof NavigationTab]

export const tab = writable<NavigationTabType>(NavigationTab.Waypoints);
```

Consumers could then chose to use:

```ts
import { NavigationMap, NavigationTab } from '@viamrobotics/prime-blocks'

<NavigationMap tab={NavigationTab.Obstacles} />
// or
<NavigationMap tab='Obstacles' />
```

Reply:
Gotcha! Updated!